src/lib.rs.in

```diff
@@ -41,6 +40,7 @@ class DeviceSigningKeyPair {
 
 foreigner_class!(
 /// ID of a user. Unique with in a segment. Must match the regex `^[a-zA-Z0-9_.$#|@/:;=+'-]+$`.
+#[derive(Clone)]
```

Comment:
Do you understand why this is needed now but not before?

Reply:
I don't. I was getting an error that it didn't know how to make a Vec<GroupId> or Vec<UserId>, so I looked at his example tests and noticed that his foreigner classes had this so I added it in and it fixed the issue.

Reply:
This possibly means that the old version was handing back values that weren't safe for the JVM. 
Not sure how we can test this... Maybe something like: get a List back in Scala, grab a reference to one of the values and then empty the list (or set it to null, or similar) and make sure the element value still works properly.

Reply:
Here's the commit where he adds it:
https://github.com/Dushistov/rust_swig/commit/d29855ba0ef87923c186fe12aecc16d7e1329e2e